core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -89,52 +92,57 @@ public int run(final String[] args, final Properties config) {
 
         int exitCode = EXIT_CODE_SUCCESS;
 
-        AdminClient adminClient = null;
-        ZkUtils zkUtils = null;
+        KafkaAdminClient kafkaAdminClient = null;
+
         try {
             parseArguments(args);
             dryRun = options.has(dryRunOption);
 
-            adminClient = AdminClient.createSimplePlaintext(options.valueOf(bootstrapServerOption));
             final String groupId = options.valueOf(applicationIdOption);
 
+            validateNoActiveConsumers(groupId);
 
-            zkUtils = ZkUtils.apply(options.valueOf(zookeeperOption),
-                30000,
-                30000,
-                JaasUtils.isZkSecurityEnabled());
+            Properties adminClientProperties = new Properties();
+            adminClientProperties.put("bootstrap.servers", options.valueOf(bootstrapServerOption));
+            kafkaAdminClient = (KafkaAdminClient) AdminClient.create(adminClientProperties);
 
             allTopics.clear();
-            allTopics.addAll(scala.collection.JavaConversions.seqAsJavaList(zkUtils.getAllTopics()));
-
-
-            if (!adminClient.describeConsumerGroup(groupId, 0).consumers().get().isEmpty()) {
-                throw new IllegalStateException("Consumer group '" + groupId + "' is still active. " +
-                            "Make sure to stop all running application instances before running the reset tool.");
-            }
+            allTopics.addAll(kafkaAdminClient.listTopics().names().get(60, TimeUnit.SECONDS));
```

Comment:
Could `listTopics` throw some TimeoutException? Do we need to handle such cases? cc @cmccabe

Reply:
The `get()` method can, but it is handled by line 118.